### PR TITLE
Fix var match

### PR DESCRIPTION
**Pull request by Maninda**
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java [hunk 1]
```diff
@@ -61,4 +68,35 @@ public String generateScriptForOperation(Class<?> generatorClass, List<DMVariabl
...
+					typeConvertedPrettyVariable = prettyVariable;
+					log.warn("Unidentified type conversion was detected from " + outputDataType.toString() + " to "
+							+ inputDataType.toString() + ".");
+				}
+			} else {
+				typeConvertedPrettyVariable = prettyVariable;
+			}
+			return typeConvertedPrettyVariable;
+		} catch (DataMapperException e) {
+			throw new IllegalArgumentException("Unknown MappingConfigGenerator type found.");
```
Comment (madushadhanushka) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: Exception e is ignored

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java [hunk 2]
```diff
@@ -19,20 +19,27 @@
 import java.util.Map;
 import java.util.Stack;
 
+import org.apache.commons.logging.Log;
 import org.wso2.developerstudio.datamapper.SchemaDataType;
 import org.wso2.developerstudio.datamapper.diagram.custom.generator.ForLoopBean;
+import org.wso2.developerstudio.datamapper.diagram.Activator;
```
Comment (madushadhanushka) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: We can reorder this import

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java (hunk 1 above)
Comment (Maninda) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: Corrected

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java (hunk 2 above)
Comment (Maninda) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: Corrected

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java [hunk 3]
```diff
@@ -61,4 +66,35 @@ public String generateScriptForOperation(Class<?> generatorClass, List<DMVariabl
...
+	private String appendTypeCorrectedInputVariable(StringBuilder operationBuilder, List<DMVariable> inputVariables, Map<String, List<SchemaDataType>> variableTypeMap,
+			Stack<ForLoopBean> parentForLoopBeanStack, List<ForLoopBean> forLoopBeanList, Map<String, Integer> outputArrayVariableForLoop,
+			Map<String, Integer> outputArrayRootVariableForLoop, SchemaDataType outputDataType) {
+		
+		try {
+			String prettyVariable = ScriptGenerationUtil.getPrettyVariableNameInForOperation(inputVariables.get(0), variableTypeMap,
+				parentForLoopBeanStack, true, forLoopBeanList, outputArrayVariableForLoop, outputArrayRootVariableForLoop);
+			SchemaDataType inputDataType = inputVariables.get(0).getSchemaVariableType();
+			String typeConvertedPrettyVariable = "";
+			if(!outputDataType.equals(inputDataType)) {
```
Comment (malakaganga) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: Formatting issue shall we fix this and other places as well ?

Comment (Maninda) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java]: Corrected

**Comment by Maninda** [comment]
Merged

**Pull request merged**